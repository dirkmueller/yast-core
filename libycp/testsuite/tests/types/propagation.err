Parsed:
----------------------------------------------------------------------
{
    // integer i
    // float fi
    // float f
    // filename: "tests/types/propagation.ycp"
    integer i = 5;
    float fi = /* integer -> float */i;
    float f = /* const integer -> float */5;
    return [f, fi];
}
----------------------------------------------------------------------
[Interpreter] tests/types/propagation.ycp:16 Implicit float conversion will loose accuracy
Parsed:
----------------------------------------------------------------------
{
    // integer i
    // float fi
    // float f
    // filename: "tests/types/propagation.ycp"
    integer i = 5;
    float fi = /* integer -> float */i;
    float f = /* const integer -> float */5;
    i = /* float -> integer */f;
    return [f, fi];
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // filename: "tests/types/propagation.ycp"
    if ((0.5 < /* const integer -> const float */1))
    return true;
    return false;
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // filename: "tests/types/propagation.ycp"
    if ((/* const integer -> const float */1 < 0.5))
    return false;
    return true;
}
----------------------------------------------------------------------
[Interpreter] tests/types/propagation.ycp:33 Implicit float conversion will loose accuracy
Parsed:
----------------------------------------------------------------------
{
    // integer i
    // float f
    // filename: "tests/types/propagation.ycp"
    integer i = /* const float -> integer */3.14;
    float f = /* const integer -> float */3;
    return [i, f];
}
----------------------------------------------------------------------
