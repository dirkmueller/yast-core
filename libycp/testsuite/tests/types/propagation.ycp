{
  // type propagation integer -> float

  integer i = 5;
  float fi = i;
  float f = 5;
  return [f,fi];
}

{
  // type propagation integer -> float

  integer i = 5;
  float fi = i;
  float f = 5;
  i = f;	// error !
  return [f,fi];
}


{
  // #21769
  if (0.5 < 1) return true;
  return false;
}
{
  // #21769
  if (1 < 0.5) return false;
  return true;
}

{
   integer i = 3.14;
   float f = 3;
   return [i, f];
}
