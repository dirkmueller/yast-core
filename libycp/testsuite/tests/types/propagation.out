([5., 5.])
([5., 5.])
(true)
(true)
([3, 3.])
